include/sealib/compactarray.h

```diff
@@ -0,0 +1,48 @@
+#ifndef SEALIB_COMPACTARRAY_H_
+#define SEALIB_COMPACTARRAY_H_
+
+#include <math.h>
+#include <third-party/boost/include/boost/dynamic_bitset.hpp>
+
+typedef boost::dynamic_bitset<> Group;
+
+/**
+ * This compact array stores a given number of values in as low space as
+ * possible.
+ * Therefore an "epsilon" parameter is used: the smaller epsilon is, the larger
+ * the storage groups will be.
+ * @author Simon Heuser
+ */
+namespace Sealib {
+class CompactArray {
+ private:
+  std::out_of_range OUTOFBOUNDS = std::out_of_range("index out of bounds");
```

Comment:
Do you require private attributes in this header file?

Reply:
Maybe. Even though it is not accessible from the outside, when somebody experiments with the code it could be interesting to see where an `index out of bounds` occurs.